src/.server/donation-messages.ts

```diff
@@ -0,0 +1,37 @@
+import type {
+  DonationMessage,
+  GSI1 as Index,
+} from "@better-giving/donation/donation-message";
+import { tables } from "@better-giving/types/list";
+import { QueryCommand, apes } from "./aws/db";
+import { env } from "./env";
+
+export const getDonationMessages = async (
+  id: number | string,
+  nextKey: string | null
+) => {
+  const recipientId: Index.Key["gsi1PK"] = `Recipient#${id}#${env}`;
+
+  const cmd = new QueryCommand({
+    TableName: tables.donation_messages,
+    IndexName: "recipient-env-gsi" satisfies Index.Name,
+    KeyConditionExpression: "gsi1PK = :gsi1PK",
+    ExpressionAttributeValues: { ":gsi1PK": recipientId },
+    ScanIndexForward: false,
+    Limit: 10,
+    ExclusiveStartKey: nextKey
+      ? JSON.parse(Buffer.from(nextKey, "base64").toString())
+      : undefined,
+  });
+  const { Items, LastEvaluatedKey } = await apes.send(cmd);
+
+  const items = (Items || []) as DonationMessage.DBRecord[];
+  const nextPageKey = LastEvaluatedKey
+    ? Buffer.from(JSON.stringify(LastEvaluatedKey)).toString("base64")
+    : undefined;
+
+  return {
+    items: items.map(({ PK, SK, gsi1PK, gsi1SK, ...i }) => i),
```

Comment:
```suggestion
    items: items.map(({ PK, SK, gsi1PK, gsi1SK, donor_id, ..i }) => i),
```
we should make sure not to expose user email addresses - but don't worry about the client, it is handled in separate PR #3593